carbon-mb/components/transports/mqtt/src/main/java/org/wso2/carbon/andes/transports/config/YAMLTransportConfigurationBuilder.java

```diff
@@ -33,14 +36,28 @@
 
     public static final String MQTT_TRANSPORT_CONF = "transport.mqtt.conf";
 
+    private static final Log log = LogFactory.getLog(YAMLTransportConfigurationBuilder.class);
+
     public static MqttTransportConfiguration readConfiguration() throws FileNotFoundException {
 
-        String mqttTransportConfigPath = System.getProperty(MQTT_TRANSPORT_CONF, "conf" + File
-                .separator + "transports" + File.separator + "mqtt-transports.yaml");
-        InputStream input = new FileInputStream(new File(mqttTransportConfigPath));
-        Yaml yaml = new Yaml();
-        yaml.setBeanAccess(BeanAccess.FIELD);
-        MqttTransportConfiguration mqttConfiguration = yaml.loadAs(input, MqttTransportConfiguration.class);
+        MqttTransportConfiguration mqttConfiguration;
+        InputStream input = null;
+        try {
+            String mqttTransportConfigPath = System.getProperty(MQTT_TRANSPORT_CONF, "conf" + File
```

Comment:
could you elaborate this more ? 

this LOC would derive the configuration path and it would eventually get the files input stream. Line 48. If the file do not reside it would throw FileNotFoundException.